sphinx/writers/latex.py

```diff
@@ -1430,6 +1433,8 @@ def visit_entry(self, node):
             self.needs_linetrimming = 1
         if len(node) > 2 and len(node.astext().split('\n')) > 2:
             self.needs_linetrimming = 1
+        if len(node.traverse(nodes.paragraph)) >= 2:
+            self.table.has_oldproblematic = True
```

Comment:
i have re-inserted these lines which were removed previously. I do not understand exactly the meaning of the node test, but with this at least the column spec uses again ``\X`` specifier which makes table overflow impossible. But in case of a tabulary, not the ``\X`` but the ``T`` (aka ``J``)  column type is used, because now tabulary handles much better cells and merged cells with paragraphs.